portlayer/exec/container.go

```diff
@@ -80,44 +86,78 @@ func (c *Container) newHandle() *Handle {
 	return newHandle(c)
 }
 
-func (c *Container) Commit(ctx context.Context, sess *session.Session, s *types.VirtualMachineConfigSpec) error {
+func (c *Container) cacheExecConfig(ec *metadata.ExecutorConfig) {
 	c.Lock()
 	defer c.Unlock()
 
-	if c.vm != nil {
-		// reconfigure
-		_, err := tasks.WaitForResult(ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
-			return c.vm.Reconfigure(ctx, *s)
-		})
+	c.ExecConfig = ec
+}
 
-		if err != nil {
-			return err
-		}
-	} else {
-		// Find the Virtual Machine folder that we use
-		folders, err := sess.Datacenter.Folders(ctx)
-		if err != nil {
-			return err
+func (c *Container) Commit(ctx context.Context, sess *session.Session, h *Handle) error {
+	c.Lock()
```

Comment:
I believe this is only a problem if you are passing the struct by value, so of course the lock state will be invalid in the copy. Here we are not doing that, `Container` is always used as a pointer.